python/ql/lib/semmle/python/ApiGraphs.qll

```diff
@@ -474,6 +492,49 @@ module API {
     int getNumArgument() { result = count(this.getArg(_)) }
   }
 
+  /**
+   * A read of a subscript that is connected to the API graph.
+   */
+  class SubscriptReadNode extends DataFlow::Node {
+    API::Node subscripted;
+
+    SubscriptReadNode() { this = subscripted.getASubscript().asSource() }
+
+    /** Gets ans API node for the object being subscripted */
+    API::Node getObject() { result = subscripted }
+
+    /** Gets the data flow node representing the index of this read, if any. */
```

Comment:
I'm surprised by the `if any` part. Will it ever be the case that there is no result for this predicate? :open_mouth: -- that is, a subscript without an index? :open_mouth:

Reply:
Hm, I probably mean "if we can find it by assuming a `SubscriptNode` is here" but that is not what is expressed. I agree the current text is confusing. Perhaps "if it can be determined"?

Reply:
I have just removed the "if any". At the moment there will always be a result. If we add more ways of reading or writing subscripts, we will probably do it in a `DataFlow::SubscriptNode` and it will become apparent if the index can be determined or not. If not, we can update these comments.